generators/app/templates/.gitignore

```diff
@@ -1,35 +1,49 @@
-################################################
-# macOS
-#################################################
-*.DS_Store
-.AppleDouble
-.LSOverride
+# System
+# ---------------------------
+.DS_Store
+Thumbs.db
 
-# Icon must end with two \r
-Icon
+# Config
+# ---------------------------
+*.esproj
+*.sublime-project
+*.sublime-workspace
+*.tmproj
+*.tmproject
+.env
+.env.php
+.idea
+.project
+.settings
 
-# Thumbnails
-._*
+# Compiled Files
+# ---------------------------
+/dist/
 
-# Files that might appear in the root of a volume
-.DocumentRevisions-V100
-.fseventsd
-.Spotlight-V100
-.TemporaryItems
-.Trashes
-.VolumeIcon.icns
-.com.apple.timemachine.donotpresent
+# Craft
+# ---------------------------
+/craft/plugins/*
+/craft/rev-manifest.json
+/craft/storage/*
 
-# Directories potentially created on remote AFP share
-.AppleDB
-.AppleDesktop
-Network Trash Folder
-Temporary Items
-.apdisk
+# Craft Public
+# ---------------------------
+/public/*
```

Comment:
Figured it was easier to just ignore everything in `/public/` except for the things we need. The downside here is that for localized sites we'll have to remember to put the localization folders in here.